treehouses remote help json (fixes #1241)

kjong opened the pull request:
Fixes #1241

Takes a bit under 3 seconds on my machine.

Line breaks are included in the descriptions.

LordJashin32 commented:
Wonder if we should keep anime.sh, margaritas, and etc out of this but i guess fine for now.

kjong commented:
```
{
	"help": "Usage: treehouses\n help [command] gives you a more detailed info about the command or will output this\n verbose <on|off> makes each command print more output (might not work with treehouses remote)\n expandfs expands the partition of the RPI image to the maximum of the SDcard\n rename <hostname> changes hostname\n password <password|disable|enable> changes the password for 'pi' user or disables/enables password authentication\n sshkey <add|list|delete|deleteall|github> used for adding or removing ssh keys for authentication\n version [contributors] returns the version of treehouses command\n image returns version of the system image installed\n detect [bluetooth|rpi|arm|wifi] detects the hardware version of any device\n ethernet <ip> <mask> <gateway> <dns> configures rpi network interface to a static ip address\n discover <scan|interface|ping|ports|mac> performs network scan and discovers all raspberry pis on the network\n <rpi> [ipaddress|url|macaddress]\n <wifi>\n wifi <ESSID> [password] connects to a wifi network\n wifihidden <ESSID> [password] connects to a hidden wifi network\n staticwifi <ip> <mask> <gateway> <dns> configures rpi wifi interface to a static ip address\n <ESSID> [password]\n wifistatus displays signal strength in dBm and layman nomenclature\n bridge <ESSID> <hotspotESSID> configures the rpi to bridge the wlan interface over a hotspot\n [password] [hotspotPassword]\n config [update|add|delete|clear] commands for interacting with config file\n container <none|docker|balena> enables (and start) the desired container\n bluetooth [on|off|pause|restart|button] switches bluetooth from regular to hotspot mode and shows id or MAC address\n [mac|id|status|log]\n ap <local|internet> <ESSID> [password] creates a mobile ap, which has two modes: local (no eth0 bridging), internet (eth0 bridging)\n aphidden <local|internet> <ESSID> creates a hidden mobile ap, with or without internet access\n [password]\n apchannel [channel] sets or prints the current ap channel\n timezone <timezone> sets the timezone of the system\n locale <locale> sets the system locale\n ssh <on|off> enables or disables the ssh service\n vnc [on|off|info] enables or disables the vnc server service\n default sets a raspbian back to default configuration\n wificountry <country> sets the wifi country\n upgrade [tag|check|bluetooth|force|cli] upgrades treehouses package using npm\n sshtunnel <add|remove|list|check|notice> helps adding an sshtunnel\n <key|portinterval> [user@host]\n led [green|red] [mode] sets the led mode\n [dance|thanksgiving|christmas]\n [newyear|lunarnewyear|valentine]\n [carnival|stpatricks|random] \n power [status|freq|<mode>] sets the power scaling or check CPU frequency\n rtc <on|off> [rasclock|ds3231] sets up the rtc clock specified\n ntp <local|internet> sets rpi to host timing locally or to get timing from a remote server\n networkmode outputs the current network mode\n button <off|bluetooth> gives the gpio pin 18 an action\n feedback <message> sends feedback\n clone [device path] clones the current SDCard onto a secondary SDCard or specified device\n restore [device path] restores a treehouses image to an SDCard or specified device\n burn [device path] download and burns the latest treehouses image to the SDcard or specified device\n rebootneeded shows if reboot is required to apply changes\n reboots <now|in|cron> reboots at given frequency | removes it if reboot task active\n <daily|weekly|monthly>\n internet checks if the rpi has access to internet\n services executes the given command on the specified service\n <available|installed [full]> list available, and installed services\n <ports|running [full]> list ports used by services and services running\n [planet] <install> Planet Learning is a generic learning system built in Angular & CouchDB\n [kolibri] <up> Kolibri is a learning platform using DJango\n [nextcloud] <start> Nextcloud is a safe home for all your data, files, etc\n [netdata] <restart> Netdata is a distributed, real-time performance and health monitoring for systems\n [mastodon] <stop> Mastodon is a free, open-source social network server\n [moodle] <down> Moodle is a learning management system built in PHP\n [pihole] <cleanup> Pi-hole is a DNS sinkhole that protects your devices from unwanted content\n [privatebin] <autorun [true|false]> PrivateBin is a minimalist, open source online pastebin\n [portainer] <url [local|tor]> Portainer is a lightweight management UI for Docker environments\n [ntopng] <info> Ntopng is a network traffic probe that monitors network usage\n [couchdb] <size> CouchDB is an open-source document-oriented NoSQL database, implemented in Erlang\n [mariadb] <icon> MariaDB is a community-developed fork of the MySQL relational database management system\n [mongodb] <port|ps> MongoDB is a general purpose, distributed, document-based, NoSQL database\n [seafile] <environment [edit]> Seafile is an open-source, cross-platform file-hosting software system\n [turtleblocksjs] TurtleBlocks is an activity with a Logo-inspired graphical \"turtle\" that draws colorful art\n [musicblocks] MusicBlocks is a programming language for exploring musical concepts in an fun way\n [minetest] Minetest is an open source infinite-world block sandbox game engine with survival and crafting\n tor [list|add|delete|deleteall|start] deals with services on tor hidden network\n [stop|destroy|notice|status|refresh]\n bootoption <console|desktop> [autologin] sets the boot mode\n <modules|params>\n openvpn [use|show|delete] helps setting up an openvpn client\n [notice|start|stop|load]\n coralenv [install|demo-on|demo-off] plays with the coral environmental board\n [demo-always-on]\n memory [total|used|free] displays the total memory of the device, the memory used as well as the available free memory\n temperature [celsius|fahrenheit|kelvin] displays raspberry pi's CPU temperature\n speedtest tests internet download and upload speed\n camera [on|off|capture] enables camera, disables camera, captures png photo\n cron [list|add|delete|deleteall] adds, deletes a custom cron job or deletes, lists all cron jobs\n [0W|tor|timestamp] adds premade cron job (or removes it if already active)\n usb [on|off] turns usb ports on or off\n remote [check|status|upgrade|services] helps with treehouses remote android app\n [version|commands|allservices]\n log <0|1|2|3|4|show|max> gets/sets log level and shows log\n blocker <0|1|2|3|4||max> website blocking levels using /etc/hosts\n sdbench displays read and write speed of micro SD card\n inspire [joke|qotd|random] displays quote based on user input\n shutdown [now|in|force] shutdown the system",
	"anime": "\nUsage: cli.sh anime\n\nExample:\n cli.sh anime\n Shows the latest 25 animes, and descriptions from the last month",
	"apchannel": "\nUsage: cli.sh apchannel [channel]\n\nPrints out or sets the ap channel.\n\nExample:\n cli.sh apchannel\n This will print out the current ap channel\n\n cli.sh apchannel 6\n This will set the ap channel to 6.\n A reboot is required to apply the changes.",
	"aphidden": "\nUsage: cli.sh aphidden <local|internet> <ESSID> [password]\n\nWhen the Raspberry pi is connected to a network via an ethernet cable this command\ncreates a wireless access point that users can connect to via wifi. If the mode is\n'internet' the ethernet connection will be shared in the access point.\n\nExamples:\n cli.sh aphidden local apname apPassword\n Creates a hidden ap with ESSID 'apname' and password 'apPassword'.\n This hotspot will not share the ethernet connection if present.\n\n cli.sh aphidden local apname\n Creates an open hidden ap with ESSID 'apname'.\n This hotspot will not share ethernet connection when present.\n\n cli.sh aphidden internet apname apPassword\n Creates a hidden ap with ESSID 'apname' and password 'apPassword'.\n This hotspot will share the ethernet connection when present.\n\n cli.sh aphidden internet apname\n Creates an open hidden ap with ESSID 'apname'.\n This hotspot will share the ethernet connection when present.\n\n This command can be used with the argument '--ip=x.y.z.w' to specify the base ip (x.y.z) for the clients/ap.\n\n cli.sh aphidden internet apname --ip=192.168.2.24\n All the clients of this network will have an ip under the network 192.168.2.0",
	"ap": "\nUsage: cli.sh ap <local|internet> <ESSID> [password]\n\nCreates a mobile ap. If the mode is 'internet' the ethernet connection will be shared in the ap.\n\nExamples:\n cli.sh ap local apname apPassword\n Creates an ap with ESSID 'apname' and password 'apPassword'.\n This hotspot will not share the ethernet connection if present.\n\n cli.sh ap local apname\n Creates an open ap with ESSID 'apname'.\n This hotspot will not share ethernet connection when present.\n\n cli.sh ap internet apname apPassword\n Creates an ap with ESSID 'apname' and password 'apPassword'.\n This hotspot will share the ethernet connection when present.\n\n cli.sh ap internet apname\n Creates an open ap with ESSID 'apname'.\n This hotspot will share the ethernet connection when present.\n\n This command can be used with the argument '--ip=x.y.z.w' to specify the base ip (x.y.z) for the clients/ap.\n\n cli.sh ap internet apname --ip=192.168.2.24\n All the clients of this network will have an ip under the network 192.168.2.0",
	"blocker": "\nUsage: cli.sh blocker <0|1|2|3|4|max>\n\nExample:\n cli.sh blocker\n blocker 0: blocker is disabled\n\n cli.sh blocker 0\n blocker 0: blocker disabled\n\n cli.sh blocker 1\n blocker 1: level set to ads (adware + malware)\n\n cli.sh blocker 2\n blocker 2: level set to ads + porn\n\n cli.sh blocker 3\n blocker 3: level set to ads + gambling + porn\n\n cli.sh blocker 4\n blocker 4: level set to ads + fakenews + gambling + porn\n\n cli.sh blocker max\n blocker X: level set to ads + fakenews + gambling + porn + social",
	"bluetooth": "\nUsage: cli.sh bluetooth [on|off|pause|restart|mac|id|button|status|log]\n\nSwitches between hotspot / regular bluetooth mode, or displays the bluetooth mac address\n\nExample:\n cli.sh bluetooth\n on\n\n cli.sh bluetooth status\n bluetooth service status: on\n rpibluetooth service status: on\n\n cli.sh bluetooth on\n This will start the bluetooth server, which lets the user control the raspberry pi using the mobile app.\n\n cli.sh bluetooth off\n This will stop the bluetooth server, and bring everything back to regular mode.\n This will also remove the bluetooth device id.\n\n cli.sh bluetooth pause\n Performs the same as 'cli.sh bluetooth off'\n The only difference is that this command will not remove the bluetooth device id.\n\n cli.sh bluetooth restart\n This will restart the bluetooth server using cli.sh bleutooth 'off' and 'on'\n\n cli.sh bluetooth mac\n This will display the bluetooth MAC address\n\n cli.sh bluetooth id\n This will display the network name along with the bluetooth id number\n\n cli.sh bluetooth button\n When the GPIO pin 18 is on the bluetooth will ne turned off\n Otherwise the bluetooth mode will be changed to hotspot\n\n cli.sh bluetooth id number\n This will display the bluetooth id number\n\n cli.sh bluetooth log\n This will display the logs of bluetooth services\n\n cli.sh bluetooth log follow\n This will display the logs as they come in live of the bluetooth services\n press (ctrl + c) to exit",
	"bootoption": "\nUsage: cli.sh bootoption <modules|params|console|console autologin|desktop|desktop autologin>\n\nChanges the boot mode, to console or desktop\n\nExample:\n cli.sh bootoption console\n The rpi will boot to console by default\n\n cli.sh bootoption console autologin\n The rpi will boot to console by default and autologin in the user that run the command\n\n cli.sh bootoption desktop\n The rpi will boot to desktop by default\n\n cli.sh bootoption desktop autologin\n The rpi will boot to desktop by default and autologin in the user that run the command\n\n cli.sh bootoption params\n Shows parameters the system booted with\n\n cli.sh bootoption modules\n Shows modules the kernel loaded at boot",
	"bridge": "\nUsage: cli.sh bridge <WifiESSID> <HotspotESSID> [WifiPassword] [HotspotPassword]\n\nBridges the wlan0 interface to ap0, creating a hotspot with desired configuration.\n\nExamples:\n cli.sh bridge MyWifi MyHotspot\n This will connect to 'MyWifi' which is an open essid, and create an open hotspot called 'MyHotspot'\n\n cli.sh bridge MyWifi MyHotspot 12345678\n This will connect to 'MyWifi' which has a password '12345678', and create an open hotspot called 'MyHotspot'\n\n cli.sh bridge MyWifi MyHotspot 12345678 hotspot123\n This will connect to 'MyWifi' which has a password '12345678', and create a password hotspot called 'MyHotspot' with the password 'hotspot123'\n\n cli.sh bridge MyWifi MyHotspot \"\" 12345678\n This will connect to 'MyWifi' which is an open essid, and create a password hotspot called 'MyHotspot' with the password '12345678'\n\n This command can be used with the argument '--ip=x.y.z.w' to specify the base ip (x.y.z) for the clients/ap.\n\n cli.sh bridge MyWifi MyHotspot \"\" 12345678 --ip=192.168.2.2\n All the clients of this network will have an ip under the network 192.168.2.0",
	"burn": "\nUsage: cli.sh burn [device path]\n\ndownloads and burns the treehouse image to the specified device\n\nExample:\n cli.sh burn\n Will download or update the treehouses image and write it to /dev/sdb (by default).\n\n cli.sh burn /dev/sda\n Will download or update the treehouses image and write it to /dev/sda",
	"button": "\nUsage: cli.sh button <off|bluetooth>\n\nGives the gpio pin 18 an action.\n Pin 1 Pin2\n +3V3 [ ] [ ] +5V\n SDA1 / GPIO 2 [ ] [ ] +5V\n SCL1 / GPIO 3 [ ] [ ] GND\n GPIO 4 [ ] [ ] GPIO 14 / TXD0\n GND [ ] [ ] GPIO 15 / RXD0\n GPIO 17 [ ] [ ] GPIO 18\n GPIO 27 [ ] [ ] GND\n GPIO 22 [ ] [ ] GPIO 23\n +3V3 [X] [X] GPIO 24\n MOSI / GPIO 10 [ ] [ ] GND\n MISO / GPIO 9 [ ] [ ] GPIO 25\n SCLK / GPIO 11 [ ] [ ] GPIO 8 / CE0#\n GND [ ] [ ] GPIO 7 / CE1#\nID_SD / GPIO 0 [ ] [ ] GPIO 1 / ID_SC\n GPIO 5 [ ] [ ] GND\n GPIO 6 [ ] [ ] GPIO 12\n GPIO 13 [ ] [ ] GND\n MISO / GPIO 19 [ ] [ ] GPIO 16 / CE2#\n GPIO 26 [ ] [ ] GPIO 20 / MOSI\n GND [ ] [ ] GPIO 21 / SCLK\n Pin 39 Pin 40\n\n\nExamples:\n cli.sh button off\n Disables the action that is run when the GPIO pin 18 is on\n\n cli.sh button bluetooth\n When the GPIO pin 18 is on the bluetooth will be turned off.\n Otherwise the bluetooth mode will be changed to hotspot.",
	"camera": "\n Usage: cli.sh camera [on|off|capture] enables camera, disables camera, captures png photo\n\n Example:\n cli.sh camera\n Config file has Camera settings which are currently disabled. Use \"cli.sh help camera\" for more commands.\n\n cli.sh camera on\n Camera is already enabled. Use \"cli.sh camera capture\" to take a photo.\n If you are having issues using the camera, try rebooting.\n\n cli.sh camera off\n Camera has been disabled. Reboot needed for settings to take effect.\n\n cli.sh camera capture\n Camera is capturing and storing a time-stamped photo in .",
	"clone": "\nUsage: cli.sh clone [device path]\n\nclones your treehouses image to an SDCard\n\nExample:\n cli.sh clone\n Will clone the current system to /dev/sdb (by default).\n\n cli.sh clone /dev/sda\n Will clone the current system to /dev/sda",
	"config": "\nUsage: cli.sh config [update|add|delete|clear]\n\ncommands for interacting with config file\n\nExample:\n cli.sh config\n Print list of config variables and values\n\n cli.sh config update varname value\n update value of variable in config file\n\n cli.sh config add varname value\n adds variable to config file\n\n cli.sh config delete varname\n removes variable from config file\n\n cli.sh config clear\n deletes config file",
	"container": "\nUsage: cli.sh container <docker|balena|none>\n\nStarts the desired container.\n\nExample:\n cli.sh container\n This will identify whether docker, balena or none of the services is currently running.\n\n cli.sh container docker\n This will start and enable the docker service. The balena service will be stopped and disabled.\n\n cli.sh container balena\n This will start and enable the balena service. The docker service will be stopped and disabled.\n\n cli.sh container none\n This will stop and disable the balena and docker service.",
	"coralenv": "\nUsage: cli.sh coralenv <demo-on|demo-always-on|demo-off|help>\n\nControls the Environmental Board\n\nExample:\n cli.sh coralenv install\n Installs the for coral environmental board necessary python packages.\n\n cli.sh coralenv demo-on\n Starts the Coral Environmental board demo.\n The Coral Environmental board will be displaying sensor information.\n The board's display will turn off on reboot.\n\n cli.sh coralenv demo-always-on\n Starts the Coral Environmental board demo.\n The board's display will persist on reboot.\n\n cli.sh coralenv demo-off\n Stops the demo and the board's display will turn off on reboot.",
	"cron": "\nUsage: cli.sh cron [options] lists all active cron jobs [adds job to cron, or removes it if present]\n\noptions:\n list lists all cron jobs\n add <\"cron job\"> see \"HOW TO ADD A CRON JOB\" below\n delete <\"word/phrase\"> deletes the cron job matching the word or phrase in quotes\n deleteall deletes all cron jobs\n 0W creates a daily reboot task (for RPi Zero W)\n tor sends \"tor notice now\" every 72 hours\n timestamp creates /var/log/uptime.log, logging every 15 minutes\n\n\nHOW TO ADD A CRON JOB:\n set the frequency (shown below) of when you want the job to execute followed by the command\n example: cli.sh cron add \"*/15 6-8 cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile 3 cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile uname\"\n This will execute \"uname\" at every 15th minute within the hours of 6 through 8 in March\n Frequency Method: cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile command-to-execute\n │ │ │ │ │\n │ │ │ │ └── day of the week (* | #/# | 0 - 6 or Sun to Sat)\n │ │ │ └──── month (* | #/# | 1 - 12)\n │ │ └────── day of the month (* | #/# | 1 - 31)\n │ └──────── hour (* | #/# | 0 - 23)\n └────────── minute (* | #/# | 0 - 59)\n\nexamples:\n cli.sh cron\n List of cron jobs:\n 0 */72 cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile treehouses tor notice now\n */15 cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile date >> /var/log/uptime.log\n @daily reboot\n\n cli.sh cron 0W\n \"daily reboot\" cron job removed\n\n cli.sh cron delete \"uname -n\"\n cron job(s) containing \"uname -n\" deleted",
	"default": "\nUsage: cli.sh default [network]\n\nResets the raspberry pi to default.\nYou can also just default the network by specifying it.\n\nExample:\n cli.sh default\n This will allow you to return back to the original configuration for all the services and settings which were set for the image when it was first installed.\n This will not delete any new files you created.\n\n cli.sh default network\n This will return the network back to the original configuration of when installed.\n This will not delete any new files you created.\n\n cli.sh default tunnel\n This will return the tunnel back to the original configuration of when installed.\n This will not delete any new files you created.\n\n cli.sh default notice\n This will return the message back to its original configuration of when installed.\n This will not delete any new files you created.",
	"detect": "\nUsage: cli.sh detect\n\nDetects and outputs the hardware info\n\nExample:\n cli.sh detect\n Prints the hardware info\n\n cli.sh detect rpi\n Prints the Raspberry Pi name\n\n cli.sh detect rpi model\n Prints the model number of the RPi\n\n cli.sh detect rpi full\n Prints the full model of the RPi\n\n cli.sh detect arm\n Prints arm version\n\n cli.sh detect bluetooth\n true\n\n cli.sh detect wifi\n true",
	"discover": "\nUsage: cli.sh discover <rpi|wifi|scan|hostinterface|ping|ports ip|ping|ports[ipaddress|url]>\n\nScans the network provdied and shows the open ports. Can scan for all raspberry pis on the network as well.\n\nExample:\n cli.sh discover rpi\n Detects raspberry pis on the network.\n cli.sh discover wifi\n Detects devices hosting wifi.\n cli.sh discover scan 192.168.7.149\n Performs a network scan of the provided ip address.\n cli.sh discover scan scanme.nmap.org\n Performs a network scan of the provided url.\n cli.sh discover interface\n Displays the host interfaces and routes on the network.\n cli.sh discover ping 192.168.7.149\n Displays servers and devices running on network provided.\n cli.sh discover ports 192.168.7.149\n Displays open ports.\n cli.sh discover mac b8:29:eb:9f:42:8b \n find the ip address of mac address.",
	"ethernet": "\nUsage: cli.sh ethernet <ip> <mask> <gateway> <dns>\n\nConfigures ethernet interface (eth0) to use a static ip address\n\nExample:\n cli.sh ethernet 192.168.1.101 255.255.255.0 192.168.1.1 9.9.9.9\n Sets the ethernet interface IP address to 192.168.1.101, mask 255.255.255.0, gateway 192.168.1.1, DNS 9.9.9.9",
	"expandfs": "\nUsage: cli.sh expandfs\n\nExpands the partition of the raspberry pi image to use the whole disk\n\nExample:\n cli.sh expandfs\n The partition of the SD card in which the raspberry pi image is stored will be expanded to match the available space on the SD card after a reboot",
	"feedback": "\nUsage: cli.sh feedback <message>\n\nShares feedback with the developers\n\nExample:\n cli.sh feedback \"Hi, you are very awesome\"\n Gives some feedback that the developers will read :)",
	"image": "\nUsage: cli.sh image\n\nReturns the version of the system image which is currently running\n\nExample:\n cli.sh image\n Prints the current version of the system image.",
	"inspire": "\n Usage: cli.sh inspire [joke|qotd|random]\n\n Prints out inspirational quote based on input.\n\n Example:\n cli.sh inspire\n By believing passionately in something that does not yet exist, we create it.\n\n cli.sh inspire qotd\n If opportunity doesn’t knock, build a door.\n\n cli.sh inspire joke\n Chuck Norris once round-house kicked a salesman. Over the phone.\n\n cli.sh inspire random\n Quick decisions are unsafe decisions.",
	"internet": "\nUsage: cli.sh internet\n\nOutputs true if the rpi can reach internet, or false if it doesn't\n\nExample:\n cli.sh internet\n the rpi has access to internet -> output: true\n the rpi doesn't have access to internet -> output: false",
	"led": "\nUsage: cli.sh led [green|red] [mode]\n cli.sh led [dance|thanksgiving|christmas|newyear|lunarnewyear|valentine|carnival|stpatricks|random]\n\nSets or returns the led mode\n\nThis will help a user to identify a raspberry pi (if a user is working on many of raspberry pis)\n\n Where to find all modes: cat /sys/class/leds/led0/trigger\n\n OPTIONS OF MODES: \n default-on turns LED on\n oneshot turns LED on once\n heartbeat sets LED to heartbeat pattern\n timer sets LED to flash at a 1-second interval\n cpu0 sets LED to CPU activity\n gpio controlled through GPIO \n input under-voltage detection\n backlight turns off LED\n none sets LED to none\n kbd-[numlock|capslock|etc] sets LED when keyboard key is hit\n\nHere is the location of LEDs\n+-------------------------------------------+\n| ()2#################40() +---+\n| 1#################39 |USB|\n|#D Pi 3B/ 4B +-+ +---+\n|#I \\/ +--+ | | +---+\n|#S ()() | | CAM +-+ |USB|\n|#P () +--+ # +---+\n|#Y # +----+\n|\u001b[5m\u001b[32m[] \u001b[25m\u001b[39m +----+ # +-+ | NET|\n|\u001b[5m\u001b[31m[] \u001b[25m\u001b[39m()+---+ | | # |A| ()+------+\n+-------|PWR|------|HDMI|------|V|----------+\n +-----+ +-------+ +-----+\nExample:\n cli.sh led\n This will return the status of the green/red (if present) leds\n\n cli.sh led red\n This will return the status of the red led\n\n cli.sh led green heartbeat\n This will set the mode of the green led to heartbeat\n\n cli.sh led red default-on\n This will set the mode of the red led to default-on\n\n cli.sh led dance\n This will do a sequence with the green led\n 1 sec on; 1 off; 2 on; 1 off; 3 on; 1 off; 4 on; 1 off\n\n cli.sh led thanksgiving\n This will do a sequence with the green and red led\n\n cli.sh led christmas\n This will set the mode of the led to christmas\n\n cli.sh led newyear\n This will set the mode of the led to newyear\n\n cli.sh led valentine\n This will set the mode of the led to valentine\n\n cli.sh led lunarnewyear\n This wil set the mode of the led to lunarnewyear\n\n cli.sh led carnival\n This will set the mode of the led to carnival\n\n cli.sh led onam\n This will set the mode of the led to onam\n\n cli.sh led heavymetal\n This will set the mode of the led to heavymetal\n\n cli.sh led stpatricks\n This will set the mode of the led to stpatricks\n\n cli.sh led random\n This will set the mode of the led to one of the above festivities",
	"locale": "\nUsage: cli.sh locale <locale>\n\nSets the system locale\n\nExample:\n cli.sh locale en_US\n This will set the raspberry pi locale to en_US.\n The supported languages are in /usr/share/i18n/SUPPORTED",
	"log": "\nUsage: cli.sh log <0|1|2|3|4|show|max>\n\nExample:\n cli.sh log\n log 0: log is disabled\n\n cli.sh log 0\n log 0: log disabled\n\n cli.sh log 1\n log 1: level set to Info\n\n cli.sh log 2\n log 2: level set to Info and Warning\n\n cli.sh log 3\n log 3: level set to Info, Warning, and Error\n\n cli.sh log 4\n log 4: level set to Info, Warning, Error, and Debug\n\n cli.sh log show\n Shows the default 6 lines of log which is at (/var/log/syslog)\n @treehouses/cli: temperature fahrenheit\n\n cli.sh log show 5\n Shows 5 lines of log which is at (/var/log/syslog)\n @treehouses/cli: temperature fahrenheit\n\n cli.sh log max\n log X: level set to max",
	"memory": "\nUsage: cli.sh memory [total|used|free] [-g|-m]\n\nDisplays the various values for total, used, and free RAM memory.\n\nExample:\n cli.sh memory\n This will display in a single sentence 3 different RAM memory values in bytes for total, used and free memory.\n\n cli.sh memory -g\n This will display in a single sentence 3 different RAM memory values in gigabytes for total, used and free memory.\n\n cli.sh memory total\n This will return the numerical value for the total memory (value in bytes).\n\n cli.sh memory used\n This will return the numerical value for the used memory (value in bytes).\n\n cli.sh memory free -m\n This will return the numerical value for the remaining free memory (value in megabytes).",
	"networkmode": "\nUsage: cli.sh networkmode [info]\n\nOutputs the current network mode\n\nExample:\n cli.sh networkmode\n Will output the current network mode that has been set up using cli.sh\n\n cli.sh networkmode info\n shows the current status of the network mode",
	"ntp": "\nUsage: cli.sh ntp <local|internet>\n\nEnables or disables time through ntp servers\n\nExample:\n cli.sh ntp internet\n Configures treehouses as a client with timing sourced from the internet\n\n cli.sh ntp local\n Configures treehouses as a server with timing sourced from the onboard clock",
	"openvpn": "\nUsage: cli.sh openvpn [use|show|delete|start|stop|load]\n\nHelps setting up an openvpn client\n\nExample:\n cli.sh openvpn => shows current status\n cli.sh openvpn use <file> [password] => copies the opvn file to the right place\n cli.sh openvpn show => shows the cert \n cli.sh openvpn delete => deletes the cert\n cli.sh openvpn start => turns on the ovpn service\n cli.sh openvpn stop => turns off the ovpn service\n cli.sh openvpn load <url> [password] => downloads the cert from a server and uses it\n cli.sh openvpn notice <on|off|add|delete|list> [api_url]\n Enables or disables the propagation of the openvpn ip / status to gitter",
	"password": "\nUsage: cli.sh password <password>\n cli.sh password [disable|enable]\n\nChanges the password for 'pi' user\n\nExample:\n cli.sh password ABC\n Sets the password for 'pi' user to 'ABC'.\n\n cli.sh password disable\n Disables password authentication (only passphrase allowed)\n\n cli.sh password enable\n Enables password authentication (password or passphrase allowed)",
	"power": "Usage: cli.sh power [mode]\n cli.sh power [current|freq]\nOptions of modes:\n status returns status of power\n default ondemand mode; moves speed from min to max at about 90% load\n ondemand moves speed from min to max at about 90% load\n conservative gradually switch frequencies at about 90% load\n usespace allows any program to set CPU's frequency\n powersave all cores set at minimum frequency\n performance all cores set at maximum frequency\n\nExample:\n cli.sh power status\n This returns the status of the power\n\n cli.sh power default\n This will set the power mode to default (ondemand)\n\n cli.sh power ondemand\n This will set the power mode to ondemand\n\n cli.sh power conservative\n This will set the power mode to conservative\n\n cli.sh power usespace\n This will set the power mode to userspace\n\n cli.sh power powersave\n This will set the power mode to powersave\n\n cli.sh power performance\n This will set the power mode to performance\n\n cli.sh power current\n This will return the current power mode\n\n cli.sh power freq\n This will return the current CPU frequency",
	"rebootneeded": "\nUsage: cli.sh rebootneeded\n\nShows if a reboot is required to apply the configuration changes done by this command\n\nExample:\n cli.sh rebootneeded\n output: true",
	"reboots": "\n Usage: cli.sh reboots <now|in|cron|daily|weekly|monthly>\n\n Reboots system at selected time and removes it if reboot task already active\n\n Example:\n cli.sh reboots daily\n System will reboot daily at 00:00\n\n cli.sh reboots in 120\n System will reboot in 120 seconds. ctrl+c to cancel.\n\n cli.sh reboots cron \"0 cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile *\"\n Rebooting with frequency of \"0 cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile *\" added\n\n Set frequency: cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile cli.sh LICENSE modules package.json README.md services templates tests _treehouses Vagrantfile package.json │ │ │ │ │\n │ │ │ │ └── day of the week (* | #/# | 0 - 6 or Sun to Sat)\n │ │ │ └──── month (* | #/# | 1 - 12)\n │ │ └────── day of the month (* | #/# | 1 - 31)\n │ └──────── hour (* | #/# | 0 - 23)\n └────────── minute (* | #/# | 0 - 59)",
	"remote": "\nUsage: cli.sh remote [check | status | upgrade | services | version | commands | allservices]\n\nReturns a string representation of the current status of the Raspberry Pi\nUsed for Treehouses Remote\n\ncli.sh remote check\n<bluetooth mac> <image> <version> <detectrpi>\n │ │ │ │\n │ │ │ └── model number of rpi\n │ │ └─────────── current cli version\n │ └─────────────────────── current treehouses image version\n └──────────────────────────────────── bluetooth mac address\n\ncli.sh remote status\n<internet> <bluetooth mac> <image> <version> <detectrpi>\n │ │ │ │ │\n │ │ │ │ └── model number of rpi\n │ │ │ └───────────── current cli version\n │ │ └────────────────────── current treehouses image version\n │ └────────────────────────────────── bluetooth mac address\n └─────────────────────────────────────────────── internet connection status\n\ncli.sh remote upgrade\ntrue if an upgrade is available\nfalse otherwise\n\ncli.sh remote services [available | installed | running]\nAvailable: | Installed: | Running: <list of services>\n\ncli.sh remote version <version_number>\ntrue if <version_number> >= \"remote_min_version\" in package.json\nfalse otherwise\n\ncli.sh remote commands [json]\nreturns a list of all commands for tab completion\n\ncli.sh remote allservices\nreturns json string of services",
	"rename": "\nUsage: cli.sh rename <hostname>\n\nChanges the hostname\n\nExample:\n cli.sh rename rpi\n Sets the hostname to 'rpi'.",
	"restore": "\nUsage: cli.sh restore [device path]\n\nrestores a treehouses image to an SDCard\nand is the logical brother of 'treehouses clone'\n\nExample:\n cli.sh restore\n Will restore the current system to /dev/sdb (by default).\n\n cli.sh restore /dev/sda\n Will restore the current system to /dev/sda",
	"rtc": "\nUsage: cli.sh rtc <on|off> [ds3231|rasclock]\n\nEnables or disables the rtc clock\n\nExample:\n cli.sh rtc off\n Disables the rtc clock.\n\n cli.sh rtc on rasclock\n Set ups the system to make the 'rasclock' clock work.\n\n cli.sh rtc on ds3231\n Set ups the system to make the 'ds3231' clock work.",
	"sdbench": "\nUsage: cli.sh sdbench\n\nExample:\n cli.sh sdbench\n prints the read and write speed",
	"services": "\nAvailable Services:\n\n planet Planet Learning is a generic learning system built in Angular & CouchDB\n kolibri Kolibri is a learning platform using DJango\n nextcloud Nextcloud is a safe home for all your data, files, etc\n netdata Netdata is a distributed, real-time performance and health monitoring for systems\n mastodon Mastodon is a free, open-source social network server\n moodle Moodle is a Learning management system built in PHP\n pihole Pi-hole is a DNS sinkhole that protects your devices from unwanted content\n privatebin PrivateBin is a minimalist, open source online pastebin\n portainer Portainer is a lightweight management UI for Docker environments\n ntopng Ntopng is a network traffic probe that monitors network usage\n couchdb CouchDB is an open-source document-oriented NoSQL database, implemented in Erlang\n mariadb MariaDB is a community-developed fork of the MySQL relational database management system\n mongodb MongoDB is a general purpose, distributed, document-based, NoSQL database.\n seafile Seafile is an open-source, cross-platform file-hosting software system\n turtleblocksjs TurtleBlocks is an activity with a Logo-inspired graphical \"turtle\" \n musicblocks Music Blocks is a programming language and collection of manipulative tools for exploring musical and mathematical concepts in an integrative and fun way.\n minetest Minetest is an open source infinite-world block sandbox game engine with survival and crafting\n\n\nTop-Level Commands:\n\n Usage:\n cli.sh services available\n ..... installed [full]\n ..... running [full]\n ..... ports\n\n available lists all available services\n\n installed lists all installed services\n [full] full details\n\n running lists all running services\n [full] full details\n\n ports lists all ports used by services\n\n Examples:\n\n cli.sh services available\n\n cli.sh services running full\n\n\nService-Specific Commands:\n\n Usage:\n cli.sh services <service_name> install\n ..... up\n ..... down\n ..... start\n ..... stop\n ..... restart\n ..... autorun [true|false]\n ..... ps\n ..... url [local|tor]\n ..... port\n ..... info\n ..... size\n ..... cleanup\n ..... icon\n ..... environment [edit [vim|request|send]]\n\n install installs and pulls <service_name>\n\n up builds and starts <service_name>\n\n down stops and removes <service_name>\n\n start starts <service_name>\n\n stop stops <service_name>\n\n restart restarts <service_name>\n\n autorun outputs true if <service_name> is set to autorun or false otherwise\n [true] sets <service_name> autorun to true\n [false] sets <service_name> autorun to false\n\n ps outputs the containers related to <service_name>\n\n url lists both the local and tor url for <service_name>\n [local] lists the local url for <service_name>\n [tor] lists the tor url for <service_name>\n\n port lists the ports used by <service_name>\n\n info gives some information about <service_name>\n\n size outputs the size of <service_name>\n\n cleanup uninstalls and removes <service_name>\n\n icon outputs the svg code for the <service_name>'s icon\n\n environment outputs the contents of the .yml for <service_name> with the currently configured environment variables\n [edit] edit the .env file for <service_name>\n [vim] opens vim to edit the .env file for <service_name>\n [request] requests the command to edit the .env file for <service_name>\n [send] sends the command to edit the .env file for <service_name>\n\n Examples:\n\n cli.sh services planet up\n\n cli.sh services planet autorun\n\n cli.sh services planet autorun true\n\n cli.sh services planet url local",
	"shutdown": "\nUsage: cli.sh shutdown [now|in|force]\n\nShuts down the system\n\nExample:\n cli.sh shutdown\n System shutdown in 60 seconds. 'ctrl+c' to cancel\n\n cli.sh shutdown now\n System shutdown immediately\n\n cli.sh shutdown in <time in seconds>\n System will shutdown after the specified time. 'ctrl+c' to cancel\n\n cli.sh shutdown force\n System will force shutdown",
	"speedtest": "\nUsage: cli.sh speedtest\n\ntests internet download and upload speed\n\nExamples:\n cli.sh speedtest\n Outputs the speed of internet download and upload speed\n\n cli.sh speedtest -h\n Outputs additional speedtest options built into speedtest",
	"sshkey": "\nUsage: cli.sh sshkey <add|list|delete|deleteall|github>\n\nUsed for adding or removing ssh keys for authentication\n\nExample:\n cli.sh sshkey add \"\"\n The public key between quotes will be added to authorized_keys so user can login without password for both 'pi' and 'root' user.\n\n cli.sh sshkey list\n Will output the content of the root and pi users keys\n\n cli.sh sshkey delete \"<key>\"\n Deletes the specified public key\n\n cli.sh sshkey deleteall\n Deletes all ssh keys\n\n cli.sh sshkey github adduser|deleteuser <username>\n Downloads or deletes the public keys of the github username from/to the authorized_keys file.\n\n cli.sh sshkey github addteam <organization> <team_name> <access_token>\n Downloads the public keys of the group members and adds them to authorized_keys\n A access_token is required to make this work, it can be generated in the following link\n https://github.com/settings/tokens",
	"ssh": "\nUsage: cli.sh ssh <on|off>\n\nEnables or disables the SSH service\n\nExample:\n cli.sh ssh on\n The SSH service will be enabled. This will allow devices on your network to be able to connect to the raspberry pi using SSH.\n\n cli.sh ssh off\n The SSH service will be disabled.",
	"sshtunnel": "\nUsage: cli.sh sshtunnel <add|remove|list|key|notice> <portinterval> [user@host]\n\nHelps setting up a sshtunnel\n\nExample:\n cli.sh sshtunnel add 65400 user@server.org\n This will set up autossh with the host 'user@server.org' and open the following tunnels\n 127.0.1.1:22 -> host:65422\n 127.0.1.1:80 -> host:65480\n 127.0.1.1:2200 -> host:65482\n 127.0.1.1:4949 -> host:65449\n 127.0.1.1:5984 -> host:65484\n\n cli.sh sshtunnel remove\n This will stop the ssh tunnels and remove the extra files added\n\n cli.sh sshtunnel list\n This will output the tunneled ports and to which host\n\n cli.sh sshtunnel check\n This will run a checklist and report back the results\n\n cli.sh sshtunnel key\n This will show the public key.\n\n cli.sh sshtunnel notice <on|off|add|delete|list|now> [api_url]\n Enables or disables the propagation of the sshtunnel ports to gitter",
	"staticwifi": "\nUsage: cli.sh staticwifi <ip> <mask> <gateway> <dns> <ESSID> [password]\n\nConfigures wifi interface (wlan0) to use a static ip address\n\nExamples:\n cli.sh staticwifi 192.168.1.101 255.255.255.0 192.168.1.1 9.9.9.9 home homewifipassword\n Connects to wifi named 'home' with password 'homewifipassword' and sets the wifi interface IP address to 192.160.1.1, mask 255.255.255.0, gateway 192.168.1.1, DNS 9.9.9.9\n\n cli.sh staticwifi 192.168.1.101 255.255.255.0 192.168.1.1 9.9.9.9 home\n Connects to an open wifi named 'home' and sets the wifi interface IP address to 192.160.1.1, mask 255.255.255.0, gateway 192.168.1.1, DNS 9.9.9.9",
	"temperature": "\n Usage: cli.sh temperature [celsius|fahrenheit|kelvin]\n\n Measures CPU temperature of Raspberry Pi\n\n Example:\n cli.sh temperature\n 47.2\n\n cli.sh temperature celsius\n 47.2°C\n\n cli.sh temperature fahrenheit\n 117.0°F\n\n cli.sh temperature kelvin\n 320.35K",
	"timezone": "\nUsage: cli.sh timezone <timezone>\n\nSets the system timezone\n\nExample:\n cli.sh timezone Etc/GMT-3\n This will set the raspberry pi time to GMT-3\n When using Etc/GMT you can specify the offset, from GMT-14 up to GMT+12\n Available timezones are inside /usr/share/zoneinfo/",
	"tor": "\nUsage: cli.sh tor\n\nSetups the tor service on the rpi\n\nExamples:\n cli.sh tor\n Outputs the hostname of the tor service\n\n cli.sh tor list\n Outputs the ports that are exposed on the tor network\n Example:\n external <=> local\n 22 <=> 22\n 80 <=> 80\n the port 22 is open and routing the traffic of the local port 22,\n the port 80 is open and routing the traffic of the local port 80\n\n cli.sh tor add <port> [localport]\n Adds the desired port to be accessible from the tor network\n Redirects localport to (tor) port\n\n cli.sh tor delete <port> [localport]\n Deletes the desired port from the tor network\n\n cli.sh tor deleteall\n Deletes all local ports from the tor network\n\n cli.sh tor start\n Setups and starts the tor service\n\n cli.sh tor stop\n Stops the tor service\n\n cli.sh tor destroy\n Stops and resets the tor configuration\n\n cli.sh tor notice <on|off|now|add|delete|list> [api_url]\n Enables or disables the propagation of the tor address/ports to gitter\n\n cli.sh tor status\n Outputs the status of the tor service\n\n cli.sh tor refresh\n Creates a new tor address while keeping added ports",
	"upgrade": "\nUsage: cli.sh upgrade [check|tag|cli|bluetooth]\n\nUpgrades cli.sh package using npm\n\nExample:\n cli.sh upgrade\n This will upgrade the cli.sh package using npm and will try to install the latest version of cli.sh running on your system\n If the latest version if installed it will not try to reinstall it\n\n cli.sh upgrade tag\n This will upgrade the cli.sh package to the version with the specified tag\n\n cli.sh upgrade force\n This will upgrade the cli.sh package to the version with the -f tag\n\n cli.sh upgrade check\n checks if there is a new version of the package, outputs false if there isn't, outputs true + version if there is\n\n cli.sh upgrade bluetooth\n This will upgrade the bluetooth server to the latest if internet is available and restart bluetooth\n\n cli.sh upgrade bluetooth branchname\n This will do the same as the above but use the 'branchname' branch on the bluetooth server repository\n https://github.com/treehouses/control\n\n cli.sh upgrade cli branchname\n This will upgrade the cli to the specified 'branchname' branch on github\n https://github.com/treehouses/cli",
	"usb": "\nUsage: cli.sh usb [on|off]\n\nTurns usb ports on or off\nNote: cannot control individual usb ports\n\nExample:\n\n cli.sh usb\n Prints usb device information\n\n cli.sh usb on\n Turns the usb ports on\n\n cli.sh usb off\n Turns the usb ports off",
	"verbose": "\nUsage: cli.sh verbose <on|off>\n\nExample:\n cli.sh verbose\n Verbosity is off\n\n cli.sh verbose on\n Successfully enabled verbosity\n\n cli.sh verbose off\n Successfully disabled verbosity",
	"version": "\nUsage: cli.sh version\n\nReturns the version of cli.sh command\n\nExample:\n cli.sh version\n Prints the version of cli.sh currently installed.\n\n cli.sh version contributors\n Prints the list of github contributors for @treehouses/cli @treehouses/remote @treehouses/builder @treehouses/control",
	"vnc": "\nUsage: cli.sh vnc <on|off|info>\n\nExample:\n cli.sh vnc\n Prints the status of the VNC server (enabled or disabled).\n\n cli.sh vnc on\n The VNC service will be enabled. This will allow devices on your network to be able to connect to the raspberry pi using VNC viewer.\n This will disable html, if it is active.\n\n cli.sh vnc off\n VNC services will be disabled. echo\nEnables or disables the VNC server service\n\nExample:\n cli.sh vnc info\n Prints a detailed configuration of each required component (boot option, vnc service, x service).",
	"wificountry": "\nUsage: cli.sh wificountry <country>\n\nSets the wireless interface country. Required on rpi 3b+ in order to get wifi working.\n\nExample:\n\n cli.sh wificountry\n This will display the current wifi country setting.\n\n cli.sh wificountry US\n This will set the wifi country to 'US'.\n This configuration is used in all commands (wifi, bridge, hotspot).",
	"wifihidden": "\nUsage: cli.sh wifihidden <ESSID> [password]\n\nConnects to a hidden wifi network\n\nExample:\n cli.sh wifihidden home homewifipassword\n Connects to a hidden wifi network named 'home' with password 'homewifipassword'.\n\n cli.sh wifihidden yourwifiname\n Connects to a hidden open wifi network named 'yourwifiname'.",
	"wifi": "\nUsage: cli.sh wifi <ESSID> [password]\n\nConnects to a wifi network\n\nExample:\n cli.sh wifi home homewifipassword\n Connects to a wifi network named 'home' with password 'homewifipassword'.\n\n cli.sh wifi yourwifiname\n Connects to an open wifi network named 'yourwifiname'.",
	"wifistatus": "\n Usage: cli.sh wifistatus\n\n Displays signal strength in dBm and layman nomenclature\n\n Example:\n cli.sh wifistatus\n Error: no wifi device is present\n\n cli.sh wifistatus\n Signal strength is -40dBm\n You have a perfect signal"
}
```
